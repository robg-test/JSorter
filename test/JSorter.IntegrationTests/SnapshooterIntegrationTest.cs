using JSorter.Extensions;
using Newtonsoft.Json.Linq;
using Snapshooter.NUnit;

namespace JSorter.IntegrationTests;

[TestFixture]
public class SnapshooterIntegrationTest 
{
    //Example from: https://developer.nhs.uk/apis/gpconnect-1-6-0/accessrecord_structured_development_fhir_examples_allergies.html
    [Test]
    public void SnapshooterIntegration()
    {
        var jTest = JObject.Parse(File.ReadAllText("Json/fhirExample.json"));

        jTest = jTest.Sort();
        Snapshot.Match(jTest);
    }
}