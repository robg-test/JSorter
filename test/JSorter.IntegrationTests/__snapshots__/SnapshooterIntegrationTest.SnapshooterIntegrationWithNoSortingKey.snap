{
  "entry": [
    {
      "resource": {
        "code": [
          {
            "coding": [
              {
                "system": "https://fhir.hl7.org.uk/STU3/CodeSystem/CareConnect-SDSJobRoleName-1",
                "code": "R0260",
                "display": "General Medical Practitioner"
              }
            ]
          }
        ],
        "id": "e0244de8-07ef-4274-9f7a-d7067bcc8d21",
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-PractitionerRole-1"
          ]
        },
        "organization": {
          "reference": "Organization/db67f447-b30d-442a-8e31-6918d1367eeb"
        },
        "practitioner": {
          "reference": "Practitioner/6c41ebfd-57c3-4162-9d7b-208c171a2fd7"
        },
        "resourceType": "PractitionerRole"
      }
    },
    {
      "resource": {
        "assertedDate": "2016-02-08T10:57:34+00:00",
        "category": [
          "environment"
        ],
        "clinicalStatus": "active",
        "code": {
          "coding": [
            {
              "display": "Peanut - dietary",
              "code": "256349002",
              "system": "http://snomed.info/sct",
              "extension": [
                {
                  "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                  "extension": [
                    {
                      "url": "DescriptionID",
                      "valueId": "381852012"
                    },
                    {
                      "url": "DescriptionDisplay",
                      "valueString": "Peanut"
                    }
                  ]
                }
              ]
            }
          ],
          "text": "Nut allergy"
        },
        "criticality": "high",
        "id": "5eb0f76a-cecb-4b83-999d-ddb76e551a9b",
        "identifier": [
          {
            "system": "https://provider.nhs.uk/data-identifier",
            "value": "83426283749649"
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-AllergyIntolerance-1"
          ]
        },
        "note": [
          {
            "text": "Certainty: Certain, Severity: Life Threatening, NOTES: notes on the peanut allergy"
          }
        ],
        "patient": {
          "reference": "Patient/04603d77-1a4e-4d63-b246-d7504f8bd833"
        },
        "reaction": [
          {
            "manifestation": [
              {
                "coding": [
                  {
                    "display": "Peanut-induced anaphylaxis",
                    "code": "241933001",
                    "system": "http://snomed.info/sct",
                    "extension": [
                      {
                        "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                        "extension": [
                          {
                            "url": "DescriptionID",
                            "valueId": "362151013"
                          }
                        ]
                      }
                    ]
                  }
                ],
                "text": "Anaphalactic shock"
              }
            ],
            "severity": "severe"
          }
        ],
        "recorder": {
          "reference": "PractitionerRole/e0244de8-07ef-4274-9f7a-d7067bcc8d21"
        },
        "resourceType": "AllergyIntolerance",
        "type": "allergy",
        "verificationStatus": "unconfirmed"
      }
    },
    {
      "resource": {
        "assertedDate": "2012-05-07T01:00:00+01:00",
        "category": [
          "medication"
        ],
        "clinicalStatus": "active",
        "code": {
          "coding": [
            {
              "display": "Amoxicillin 250mg capsules",
              "code": "323509004",
              "system": "http://snomed.info/sct",
              "extension": [
                {
                  "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                  "extension": [
                    {
                      "url": "DescriptionID",
                      "valueId": "56966201000001112"
                    }
                  ]
                }
              ]
            }
          ]
        },
        "criticality": "low",
        "id": "6bff710a-0bdc-4c9b-b98b-40db0a107edc",
        "identifier": [
          {
            "system": "https://provider.nhs.uk/data-identifier",
            "value": "83426283749639"
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-AllergyIntolerance-1"
          ]
        },
        "note": [
          {
            "text": "Certainty: Likely, Severity: Minimal, NOTES: Some freetext notes"
          }
        ],
        "patient": {
          "reference": "Patient/04603d77-1a4e-4d63-b246-d7504f8bd833"
        },
        "reaction": [
          {
            "manifestation": [
              {
                "coding": [
                  {
                    "display": "O/E - itchy rash",
                    "code": "304386008",
                    "system": "http://snomed.info/sct",
                    "extension": [
                      {
                        "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                        "extension": [
                          {
                            "url": "DescriptionID",
                            "valueId": "446685017"
                          }
                        ]
                      }
                    ]
                  }
                ]
              }
            ],
            "severity": "mild"
          }
        ],
        "recorder": {
          "reference": "PractitionerRole/e0244de8-07ef-4274-9f7a-d7067bcc8d21"
        },
        "resourceType": "AllergyIntolerance",
        "type": "allergy",
        "verificationStatus": "unconfirmed"
      }
    },
    {
      "resource": {
        "assertedDate": "2016-01-08T11:57:34+00:00",
        "category": [
          "medication"
        ],
        "clinicalStatus": "active",
        "code": {
          "coding": [
            {
              "display": "Transfer-degraded drug allergy",
              "code": "196461000000101",
              "system": "http://snomed.info/sct",
              "extension": [
                {
                  "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                  "extension": [
                    {
                      "url": "DescriptionID",
                      "valueId": "294801000000114"
                    }
                  ]
                }
              ]
            }
          ],
          "text": "Allergy to paracetemol"
        },
        "id": "d92b7d42-554d-4c92-b829-e76508185702",
        "identifier": [
          {
            "system": "https://provider.nhs.uk/data-identifier",
            "value": "83426283749659"
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-AllergyIntolerance-1"
          ]
        },
        "note": [
          {
            "text": "NOTES: This was entered incorrectly on the source system as a history/journal entry and not as a drug allergy"
          }
        ],
        "patient": {
          "reference": "Patient/04603d77-1a4e-4d63-b246-d7504f8bd833"
        },
        "recorder": {
          "reference": "PractitionerRole/e0244de8-07ef-4274-9f7a-d7067bcc8d21"
        },
        "resourceType": "AllergyIntolerance",
        "type": "allergy",
        "verificationStatus": "unconfirmed"
      }
    },
    {
      "resource": {
        "code": {
          "coding": [
            {
              "display": "Resolved Allergies",
              "system": "http://snomed.info/sct",
              "code": "TBD"
            }
          ]
        },
        "contained": [
          {
            "resourceType": "AllergyIntolerance",
            "id": "Resolved-1",
            "identifier": [
              {
                "system": "https://provider.nhs.uk/data-identifier",
                "value": "83426283749629"
              }
            ],
            "extension": [
              {
                "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-CareConnect-GPC-AllergyIntoleranceEnd-1",
                "extension": [
                  {
                    "url": "endDate",
                    "valueDateTime": "2018-03-01"
                  },
                  {
                    "url": "reasonEnded",
                    "valueString": "Patient reports no subsequent recurrence on same medication"
                  }
                ]
              }
            ],
            "reaction": [
              {
                "manifestation": [
                  {
                    "coding": [
                      {
                        "display": "Vomiting",
                        "code": "422400008",
                        "system": "http://snomed.info/sct",
                        "extension": [
                          {
                            "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                            "extension": [
                              {
                                "url": "DescriptionID",
                                "valueId": "2647992016"
                              },
                              {
                                "url": "DescriptionDisplay",
                                "valueString": "Emesis"
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ],
                "severity": "severe"
              }
            ],
            "verificationStatus": "unconfirmed",
            "assertedDate": "2011-05-07T01:00:00+01:00",
            "type": "intolerance",
            "code": {
              "coding": [
                {
                  "display": "Aspirin 75mg dispersible tablet",
                  "code": "319773006",
                  "system": "http://snomed.info/sct",
                  "extension": [
                    {
                      "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
                      "extension": [
                        {
                          "url": "DescriptionID",
                          "valueId": "56962301000001119"
                        }
                      ]
                    }
                  ]
                }
              ]
            },
            "clinicalStatus": "resolved",
            "patient": {
              "reference": "Patient/04603d77-1a4e-4d63-b246-d7504f8bd833"
            },
            "note": [
              {
                "text": "Allergy Type: Adverse Reaction, Certainty: Absolute, Severity: Very Severe, NOTES: Some freetext notes"
              }
            ],
            "recorder": {
              "reference": "PractitionerRole/e0244de8-07ef-4274-9f7a-d7067bcc8d21"
            },
            "category": [
              "medication"
            ],
            "meta": {
              "profile": [
                "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-AllergyIntolerance-1"
              ]
            },
            "criticality": "low"
          }
        ],
        "date": "2018-03-01T10:57:34+00:00",
        "entry": [
          {
            "item": {
              "reference": "#Resolved-1"
            }
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-List-1"
          ]
        },
        "mode": "snapshot",
        "orderedBy": {
          "coding": [
            {
              "system": "http://hl7.org/fhir/list-order",
              "code": "event-date"
            }
          ]
        },
        "resourceType": "List",
        "status": "current",
        "subject": {
          "identifier": {
            "system": "https://fhir.nhs.uk/Id/nhs-number",
            "value": "1234567890"
          }
        },
        "title": "Resolved Allergies"
      }
    },
    {
      "resource": {
        "code": {
          "coding": [
            {
              "display": "Allergies and adverse reaction",
              "system": "http://snomed.info/sct",
              "code": "886921000000105"
            }
          ]
        },
        "date": "2018-03-01T10:57:34+00:00",
        "entry": [
          {
            "item": {
              "reference": "AllergyIntolerance/561850bd-360c-4d17-b3c8-a837ef0cbfba"
            }
          },
          {
            "item": {
              "reference": "AllergyIntolerance/5eb0f76a-cecb-4b83-999d-ddb76e551a9b"
            }
          },
          {
            "item": {
              "reference": "AllergyIntolerance/6bff710a-0bdc-4c9b-b98b-40db0a107edc"
            }
          },
          {
            "item": {
              "reference": "AllergyIntolerance/d92b7d42-554d-4c92-b829-e76508185702"
            }
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-List-1"
          ]
        },
        "mode": "snapshot",
        "orderedBy": {
          "coding": [
            {
              "system": "http://hl7.org/fhir/list-order",
              "code": "event-date"
            }
          ]
        },
        "resourceType": "List",
        "status": "current",
        "subject": {
          "reference": "Patient/04603d77-1a4e-4d63-b246-d7504f8bd833"
        },
        "title": "Active Allergies"
      }
    },
    {
      "resource": {
        "address": [
          {
            "line": [
              "Trevelyan Square",
              "Boar Ln"
            ],
            "city": "Leeds",
            "district": "West Yorkshire",
            "postalCode": "LS1 6AE"
          }
        ],
        "id": "db67f447-b30d-442a-8e31-6918d1367eeb",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/ods-organization-code",
            "value": "O001"
          }
        ],
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Organization-1"
          ],
          "versionId": "636064088098730113"
        },
        "name": "The Trevelyan Practice",
        "resourceType": "Organization",
        "telecom": [
          {
            "system": "phone",
            "value": "03003035678",
            "use": "work"
          }
        ]
      }
    },
    {
      "resource": {
        "active": true,
        "address": [
          {
            "use": "home",
            "type": "physical",
            "line": [
              "Cable Place",
              "Roundhay"
            ],
            "city": "Leeds",
            "district": "West Yorkshire",
            "postalCode": "LS1 5HT"
          }
        ],
        "birthDate": "1952-05-31",
        "extension": [
          {
            "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-CareConnect-GPC-RegistrationDetails-1",
            "extension": [
              {
                "url": "registrationPeriod",
                "valuePeriod": {
                  "start": "1962-07-13T01:00:00+01:00"
                }
              }
            ]
          }
        ],
        "gender": "female",
        "generalPractitioner": {
          "reference": "Practitioner/6c41ebfd-57c3-4162-9d7b-208c171a2fd7"
        },
        "id": "04603d77-1a4e-4d63-b246-d7504f8bd833",
        "identifier": [
          {
            "extension": [
              {
                "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-CareConnect-GPC-NHSNumberVerificationStatus-1",
                "valueCodeableConcept": {
                  "coding": [
                    {
                      "system": "https://fhir.nhs.uk/CareConnect-NHSNumberVerificationStatus-1",
                      "code": "01",
                      "display": "Number present and verified"
                    }
                  ]
                }
              }
            ],
            "system": "https://fhir.nhs.uk/Id/nhs-number",
            "value": "9999999999"
          }
        ],
        "managingOrganization": {
          "reference": "Organization/db67f447-b30d-442a-8e31-6918d1367eeb"
        },
        "meta": {
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Patient-1"
          ],
          "versionId": "1469448000000"
        },
        "name": [
          {
            "use": "official",
            "text": "JACKSON Jane (Miss)",
            "family": "Jackson",
            "given": [
              "Jane"
            ],
            "prefix": [
              "Miss"
            ]
          }
        ],
        "resourceType": "Patient",
        "telecom": [
          {
            "system": "phone",
            "value": "01454587554",
            "use": "home"
          }
        ]
      }
    },
    {
      "resource": {
        "gender": "female",
        "id": "6c41ebfd-57c3-4162-9d7b-208c171a2fd7",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/sds-user-id",
            "value": "G13579135"
          }
        ],
        "meta": {
          "lastUpdated": "2016-07-25T13:00:00+01:00",
          "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Practitioner-1"
          ],
          "versionId": "1469448000000"
        },
        "name": [
          {
            "use": "usual",
            "family": "Gilbert",
            "given": [
              "Nichole"
            ],
            "prefix": [
              "Miss"
            ]
          }
        ],
        "resourceType": "Practitioner"
      }
    }
  ],
  "meta": {
    "profile": [
      "https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-StructuredRecord-Bundle-1"
    ]
  },
  "resourceType": "Bundle",
  "type": "collection"
}
